zpay32/invoice.go

```diff
@@ -1094,6 +1094,14 @@ func writeTaggedFields(bufferBase32 *bytes.Buffer, invoice *Invoice) error {
 			return err
 		}
 	}
+	if invoice.PaymentAddr != nil {
+		err := writeBytes32(
+			bufferBase32, fieldTypeS, *invoice.PaymentAddr,
+		)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
Where exactly in the spec is this ordering requirement? I only see this:

>MUST specify the most-preferred field first, followed by less-preferred fields, in order.

But what is preferred?

Maybe add a comment that the fields needs to be order.

Reply:
Ok, misread that.